demo/b2share_demo/migration.py

```diff
@@ -27,66 +27,377 @@
+        result['resource_types'] = [{'resource_type_general': "Other"}]
+    result['community_specific'] = {}
+    if 'domain_metadata' in rec.keys():
+        result.update(
+            _match_community_specific_metadata(rec, community)
+            )
+    return result
+
+def _match_community_specific_metadata(rec, community):
+    cs_md_values_dict = rec['domain_metadata']
+    convert_to_array = [
+        'hasOntologyLanguage',
+        'usedOntologyEngineeringTool',
+    ]
+    for key in convert_to_array:
+        if key in cs_md_values_dict.keys():
+            cs_md_values_dict[key] = [
+                cs_md_values_dict[key]
+            ]
+    remove_integer_keys_with_empty_values = [
+        'planned_sampled_individuals',
+        'planned_total_individuals'
+    ]
+    for key in remove_integer_keys_with_empty_values:
+        if key in cs_md_values_dict.keys():
+            if cs_md_values_dict[key] == '':
+                cs_md_values_dict.pop(key)
+            else:
+                cs_md_values_dict[key] = int(cs_md_values_dict[key])
+    result = {}
+    result['community_specific'] = {}
+    resolve_string = "$BLOCK_SCHEMA_ID[%s]" % community.name.lower()
+    block_schema_id = resolve_block_schema_id(resolve_string)
+    result['community_specific'][block_schema_id] = cs_md_values_dict
+    return result
+
+def _save_file(logfile, old_url, filename):
+    V1_URL_BASE = current_app.config.get('V1_URL_BASE')
+    url1 = old_url.replace('https://b2share.eudat.eu/', V1_URL_BASE)
+    url = url1.replace('/api/record/', '/record/')
+
+    CHUNK_SIZE = 16 * 1024 * 1024 #download 16MB at a time
+    current_app.logger.debug("Downloading %s" % url)
+    gcontext = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
+    finished = False
+    f = open(filename, 'wb')
+    try:
+        u = urlopen(url, context=gcontext)
+    except:
+        current_app.logger.error("TIMEOUT trying to download %s" % url)
+        logfile.write("\n********************")
+        logfile.write("\nERROR: cannot open file URL for download: {}\n".format(url))
+        logfile.write(traceback.format_exc())
+        logfile.write("\n********************")
+        return
+
+    while not finished:
+        try:
+            chunk = u.read(CHUNK_SIZE)
+            if chunk:
+                f.write(chunk)
+            else:
+                finished = True
+        except:
+            logfile.write("\n********************")
+            logfile.write("\nWARN: exception while reading file: {}\n".format(url))
+            logfile.write(traceback.format_exc())
+            logfile.write("\n********************")
+            finished = True
+    f.close()
+
+
+
+def unique(lst):
+    ret = []
+    o = {}
+    for l in lst:
+        if l not in o:
+            ret.append(l)
+            o[l] = True
+    return ret
+
+assert unique([1, 1, 1, "a", 2, 2, "b", 3, "b", "a", 3]) == [1, "a", 2, "b", 3]
+assert unique([1, 2, 1, 1, 3, "a", 2, 2, "b", "b", "a"]) == [1, 2, 3, "a", "b"]
+
```

Comment:
Do these assertions belong here?
Why not return list(set(lst)) in function unique?